File: level 3/gcd_sumUptoN.cpp
#include<iostream>
using namespace std;
int gcd(int a,int b){
    if(b==0)
        return a;
    else
        return gcd(b,a%b);
}
int main(){
    int i,j,n,s=0;
    cin>>n;
    for(i=1;j<n;j++)
        for(j=i+1;j<=n;j++)
            s=s+gcd(i,j)
    cout<<s;
    return 0;
}

File: level 3/gcd_sumUptoN.py
def gcd(a,b):
    return a if(b==0) else gcd(b,a%b)
n=int(input())
s=0
for i in range(1,n):
    for j in range(i+1,n+1):
        s+=gcd(i,j)
print(s)
